### PR TITLE
Hwg170030

### PR DESCRIPTION
**Pull request by nith822**
# Fix multiple instances of userService
-Finally fix sending correct oAuthID in request body.
-"**_oAuthId**" cookie now holds the access token; "**oAuthId**" cookie has a name conflict and expires when the browsing session ends (other cookies currently last 30 days)
-"userId" cookie now holds the userID for the session

# Add retrieve loggedIn user via cookies at startup
-Login automatically if  "_oAuthId" and "userId" cookies are present

# Add username to login button if logged in already
- Login button is replaced with username if logged in
- Click on username button to logout, (no indicator currently)

# Add upload restriction by login
- Accessing the upload page while not logged in redirects to the login page with an alert

# Add viscerality changes
# More viscerality changes, lazy and put in global css
- Vist saki.rinshan.com btw

### CODE REVIEW

File: UI/src/styles.css [hunk 1]
```diff
@@ -5,3 +5,21 @@
 .label {
 	float: clear;
 }
+
+
+button, .div-translation, .div-translation-body {
+	cursor: pointer;
+}
+
+button:hover, .div-translation:hover, .div-translation-body:hover {
```
Comment (mattAbe) [inline UI/src/styles.css:14]: Replacing .div-translation-body with .div-translation-preview would highlight the whole translation including the votes

Comment (nith822) [reply UI/src/styles.css]: done

File: API/posts/PostController.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ exports.view = function (req, res, next) {
         }
         console.log('Post OP: ' + temp.userID);
         try {
-            userObjectId = mongoose.Types.ObjectId(temp.userID);
+            userObjectId = mongoose.Types.ObjectId("5e75532986559b1e8cba2959");
```
Comment (mattAbe) [inline API/posts/PostController.js:138]: Is the user id meant to be hardcoded here?

Comment (nith822) [reply API/posts/PostController.js:138]: no, this is wrong

**Pull request merged**